### DISCUSSION TITLE
Blank space bug

### DISCUSSION THREAD

**Discussion by violet-dev** [coder/code-server#3951] [Q&A] [answered]
![스크린샷, 2021-08-10 12-08-42](https://user-images.githubusercontent.com/66541802/128802388-238bae99-f990-49da-8641-744e71624187.png)

![스크린샷, 2021-08-10 12-07-42](https://user-images.githubusercontent.com/66541802/128802330-2bbfd1cf-a699-48ea-9ca3-f4ddf286c2fd.png)

![스크린샷, 2021-08-10 12-11-37](https://user-images.githubusercontent.com/66541802/128802591-5d108d1b-8674-4535-bd6c-a6a13780127f.png)

I used the 3.8 version well, then upgraded to 3.11 to use jupyter.
However, when accessing the webpage after upgrading, only a blank screen appears with just one error message.
```
GET http://13.209.66.206/vs/code/browser/workbench/workbench.js net::ERR_ABORTED 404 (Not Found)
```
What should I do?

**Comment by jsjoeio** [comment]
Can you load code-server at all? Can you try clearing your cache?

We changed the path of the service worker and forgot to update in one place. This should be fixed in 3.11.1 I believe (only out on Docker right now, but should be out on npm and homebrew soon)

If jupyter is having issues, it may be this: https://github.com/cdr/code-server/issues/3874

**Reply by jsjoeio** [reply]
Did you try upgrading to 3.11.1?

**Comment by dabiyang** [comment]
I got the same problem when standalone install ready.but not about jupyter.Just run code-server and login.blank blank blank

**Reply by jsjoeio** [reply]
Which version of code-server are you using?

**Comment by rbray89** [comment]
I'm seeing this exact issue as well.
Running behind an NGINX reverse proxy in a docker image. Works with 3.9.0.

**Reply by jsjoeio** [reply]
Which version of code-server are you using?

**Reply by rbray89** [reply]
> Which version of code-server are you using?

Tried it with 3.11.1. 3.11.0 had a similar issue. I was also able to get it working with all versions prior to 3.11.0 after 3.9.0 but was looking forward to the bugfixes in 3.11.0

**Reply by jsjoeio** [reply]
Hmm...I'm trying to think what could have caused this 🤔 I'm running 3.11.1 locally on macOS + Brave and can't reproduce. I'm not sure what this could be. cc @code-asher 

![image](https://user-images.githubusercontent.com/3806031/130528778-01e5784c-9a81-4da8-80a8-92a3b42149e5.png)

**Reply by code-asher** [reply]
I wasn't able to reproduce either:

```
$ curl -fsSL https://code-server.dev/install.sh | sh
$ code-server --version
3.11.1 c680aae973d83583e4a73dc0c422f44021f0140e
$ code-server # Browse to localhost:8080 (Chromium)
```

How was code-server 3.11.1 installed?

The problem looks to be caused by the error in `vscode.browserified.js`. It is responsible for configuring the loader paths so if it errors the workbench will fail to load.

It could be that `vscode.browserified.js` is loading incorrectly. Checking the file contents in the network tab can tell us if that's the case.

The only other thing that comes to mind is that I think we bumped the minimum JavaScript version so if this is an older browser version it might be getting tripped up on new syntax.

**Reply by rbray89** [reply]
I'm installing the deb image in an ubuntu 18.04 docker container.

Running: dumb-init code-server --disable-telemetry --disable-update-check --host 0.0.0.0 --port 8080 --auth none /workspace
Accessed behind an NGINX reverse proxy.

**Reply by rbray89** [reply]
Seems like other resources are being loaded properly. Whatever is responsible for adding the workbench.js resource isn't adding the /static/$UUID/usr/lib/code-server/lib/vscode/out/ path prefix.

**Reply by rbray89** [reply]
> I'm installing the deb image in an ubuntu 18.04 docker container.
> 
> Running: dumb-init code-server --disable-telemetry --disable-update-check --host 0.0.0.0 --port 8080 --auth none /workspace
> Accessed behind an NGINX reverse proxy.

Also just now tried without dumb-init and with only specifying host, port, and no augh with no change.

**Reply by rbray89** [reply]
> The only other thing that comes to mind is that I think we bumped the minimum JavaScript version so if this is an older browser version it might be getting tripped up on new syntax.

Using latest google-chrome-stable, so I doubt this is it.

**Reply by rbray89** [reply]
> Hmm...I'm trying to think what could have caused this 🤔 I'm running 3.11.1 locally on macOS + Brave and can't reproduce. I'm not sure what this could be. cc @code-asher
> 
> ![image](https://user-images.githubusercontent.com/3806031/130528778-01e5784c-9a81-4da8-80a8-92a3b42149e5.png)

Where is your workbench.js loading from?

**Comment by rbray89** [comment]
Additionally:

-  when I add the static prefix to the URL, I can load the workbench.js file manually. 
-  Same issue if I bypass nginx.

**Comment by rbray89** [comment]
Changing this line: 
https://github.com/cdr/code-server/blob/e1d291e59cb8a48fc9797bd4f986604f396318e4/src/browser/pages/vscode.html#L52
to add the {{CS_STATIC_BASE}} WILL load the workbench.js, but then many other subsequent resources fail to load. My guess is that the require function isn't functioning properly here.

**Comment by rbray89** [comment]
If I had to guess, I'd place my money on https://github.com/cdr/code-server/commit/be7ea8f3f76230c76bb3ea6d5c8f4bdc407a32ea#diff-1cfc145fa9fd80413296567e0be2a6c1f323e1caa379211bc9860dc09009fff4 being the culprit here.

**Comment by rbray89** [comment]
Possibly related to https://github.com/browserify/browserify/issues/1994

**Comment by rbray89** [comment]
Another datapoint, firefox is able to load the page.

**Comment by rbray89** [comment, marked as answer]
And Chromium was able to load it. Seems my google-chrome-stable apt is either out of date or something weird going on with apt, as it is stuck on 72.

Solution: update your browser.

**Reply by jsjoeio** [reply]
Interesting! Thanks for sharing your investigation notes throughout the process. Glad it was as simple as that! But yeah, not something I would have thought to do myself 😅 I figured browsers always auto-update lol

**Reply by violet-dev** [reply]
ok ill try it thank you!
